Move apps and team page scripts out of landing-page.js

amanagr opened the pull request:
Follow up on https://github.com/zulip/zulip/pull/27079#discussion_r1348730886

amanagr committed:
apps: Extract script to load apps data.

This helps us run the script independent of the URL path.

timabbott commented on web/src/page_params.ts:
```diff
@@ -1,5 +1,39 @@
+    name: string;
+} & {
+    [K in RepositoryName]?: number;
+};
```
This change is wrong -- we need a `web/src/portico/team_page_params.ts`. The page parameters sent for this other page are unrelated to those for the main web app.

Here are some examples:
```
$ ls web/src/*/*param*
web/src/billing/page_params.ts  web/src/stats/page_params.ts
```

amanagr replied:
`src/page_params` also runs for portico pages. So, I extracted a `team_page_params_types` as next best thing I could think of.

timabbott commented:
This seems like good progress, but see the note above on `page_params` types.

amanagr committed:
team: Extract script to run to its own file.

timabbott commented on web/src/page_params.ts:
```diff
@@ -9,6 +11,7 @@ export const page_params: {
         name: string;
         allowed: boolean;
     }[];
+    contributors: Contributor[] | undefined;
```
This remains wrong. The point is that `page_params` has a different type depending on what page it is. The tiny bit of code in this module that isn't declaring a type is literally one line of code. We should just put copies of that one line of code in each module that's actually using it exporting the correct types for that module's `page_params`. 

No page ever has the type you're changing this to -- the teams page only has `contributors` and maybe a couple other elements; the app doesn't have a `contributors`, and the teams page doesn't have a `max_avatar_file_size_mib`. 

(Or, better, we should just use `zod` to parse the `page_params` and do that at the top of individual modules like `teams_page.ts`, so we're actually validating the types).

amanagr committed:
google_analytics: Extract id directly from `#page-params`.

amanagr committed:
blueslip: Extract development_environment directly from `#page-params`.

amanagr committed:
sentry: Extract required params directly from `#page-params`.

amanagr committed:
sentry: Extract functions used by channel into a util file.

This is to make tests for channel work.

Channel mocks `$` to be an object, so it cannot be called as an
function in sentry.ts if it is imported in channel.ts.

So, to avoid it, we extract the functions which channel.ts uses
and fortunately they don't use `$`, so it works out.

timabbott commented on web/src/blueslip.ts:
```diff
@@ -10,7 +10,8 @@ import * as Sentry from "@sentry/browser";
 import {BlueslipError, display_stacktrace} from "./blueslip_stacktrace";
-import {page_params} from "./page_params";
+
+const development_environment = $("#page-params").data("params").development_environment;
```
I think the most likely theory for how to fix this is to put the actual app parameters into a separate `data-app-params` bucket from the "common" stuff used by Sentry/Blueslip and smaller pages, and those things can be parsed multiple times safely as they would not be huge. But I'm not sure and it deserves a proper discussion.

timabbott commented:
So the `page_params` typescript part of this was I think resolved in #28389, so this might be easier to pick apart now.

amanagr commented:
No longer relevant.
